overtime/algorithms/edgeDeletion.py

```diff
@@ -0,0 +1,285 @@
+        if root == '':
+            print('there is no node whose reachability is more than {}'.format(h))
+            break
+
+        # generate a reachable subtree based on the root
+        subtree = reachable_subtree(graph, root, h)
+
+        E_.extend(subtree.edges.uids())
+        # update (G, λ) ← (G, λ) \ E_
+        for edge in subtree.edges.uids():
+            graph.remove_edge(edge)
+
+    return E_
+
+def max_reachability(graph):
+    """
+        A method to calculate the max temporal reachability of a network.
+
+        Parameter(s):
+        -------------
+        graph : TemporalDiGraph
+                A directed, temporal graph.
+
+        Returns:
+        --------
+        maxReachability : int
+            the max temporal reachability of a network.
+
+    """
+
+    maxReachability = 0
+
+    # check the temporal reachability of each node
+    for node in graph.nodes.labels():
+        r = ot.calculate_reachability(graph, node)
+        if r > maxReachability:
+            maxReachability = r
+
+    return maxReachability
+
+
+def max_endtime(graph):
+    """
+        A method to calculate the maximum endtime of a network.
+
+        Parameter(s):
+        -------------
+        graph : TemporalDiGraph
+                A directed, temporal graph.
+
+        Returns:
+        --------
+        maxEndtime : int
+            the maximum endtime of a network.
+
+    """
+
+    maxEndtime = 0
+
+    # check the end time of each edges
+    for edge in graph.edges.set:
+        if edge.end > maxEndtime:
+            maxEndtime = edge.end
+
+    return maxEndtime
+
+
+def find_edges(graph, layout, j):
+    """
+        A method to find all the edges that span vj and vj+1.
+
+        Parameter(s):
+        -------------
+        graph : TemporalDiGraph
+                A directed, temporal graph.
+
+
+        Returns:
+        --------
+        edgeList : list
+            a list that contains all the edges that span vj and vj+1.
+
+    """
+
+    edgeList = []
+    # divided the layout into two subsets based on j
+    layout1 = layout[:j + 1]
+    layout2 = layout[j + 1:len(layout)]
+
+    # find all edges with one endpoint in {v1...vj} and one endpoint in {vj+1...vn}
+    for edge in graph.edges.set:
+        if (edge.sink.label in layout1 and edge.source.label in layout2) or (
+                edge.sink.label in layout2 and edge.source.label in layout1):
+            edgeList.append(edge.uid)
+
+    return edgeList
+
+
+def generate_Layout(graph):
+    """
+        A method to generate a layout of a network, such as {v1, v2, v3， ....， vn}.
+
+        Parameter(s):
+        -------------
+        graph : TemporalDiGraph
+                A directed, temporal graph.
+
+        Returns:
+        --------
+        layout : list
+            a layout of the network.
+
+    """
+
+    # generate a layout based on the order of their labels
+    layout = graph.nodes.labels()
+    return layout
+
+
+def c_approximation(graph, h):

```

Comment:
What is c? 

It seems that the implementation is incorrect, as it does not take into account cut-width of the graph. Read carefully the algorithm in the paper, and try to implement it correctly.